Detectors/TPC/simulation/src/SignalInduction.cxx

```diff
@@ -0,0 +1,106 @@
+  }
+  grPRFOROC12 = new TGraph2D(oroc12path.data());
+  grPRFOROC12->SetName("PRF_OROC12");
+  if (!grPRFOROC12) {
+    LOG(FATAL) << "tpc::SignalInduction - Loading OROC 1/2 PRF from " << oroc12path << " failed No PRF loaded!";
+  }
+  grPRFOROC3 = new TGraph2D(oroc3path.data());
+  grPRFOROC3->SetName("PRF_OROC3");
+  if (!grPRFOROC3) {
+    LOG(FATAL) << "tpc::SignalInduction - Loading OROC 3 PRF from " << oroc3path << " failed No PRF loaded!";
+  }
+}
+
+PadResponse SignalInduction::computePadResponse(const GlobalPosition3D posEle, const DigitPos digitPos, const double offsetPad, const double offsetRow) const
+{
+  /// The distance from the electron arrival position to the pad center is computed and the pad response function is evaluated accordingly.
+  /// The pad for which this is done can be varied with the variables offsetPad (number of pads left/right of the pad at which the electron arrives) and offsetRow (number of row above/below the row at which the electron arrives) and
+  const static Mapper& mapper = Mapper::instance();
+  const Sector sector = digitPos.getCRU().sector();
+  const GEMstack gemstack = digitPos.getCRU().gemStack();
+
+  /// Compute the cpad and global row coordinates
+  const LocalPosition3D elePosLocal = Mapper::GlobalToLocal(posEle, sector);
+  const int globalRow = digitPos.getGlobalPadPos().getRow();
+  const int globalPad = digitPos.getGlobalPadPos().getPad();
+  const int globalCPad = globalPad - 0.5 * mapper.getNumberOfPadsInRowSector(globalRow);
+
+  /// Shift by the offset and compute the offset in x and y
+  const PadPos shiftedGlobalPos(globalRow + offsetRow, globalCPad + offsetPad + 0.5 * mapper.getNumberOfPadsInRowSector(globalRow + offsetRow));
+  const GlobalPadNumber shiftedGlobalPadNum = mapper.globalPadNumber(shiftedGlobalPos);
+  const PadCentre& padCent = mapper.padCentre(shiftedGlobalPadNum);
+  const double localYfactor = (sector.side() == Side::A) ? -1.f : 1.f;
+  const double offsetX = std::abs(elePosLocal.X() - padCent.X()) * 10.f;
+  const double offsetY = std::abs(elePosLocal.Y() - localYfactor * padCent.Y()) * 10.f;
+
+  /// TODO For some reason the rows have a mismatch?
+  const DigitPos shiftedDigitPos(mapper.getCRU(sector, shiftedGlobalPadNum), mapper.padPos(shiftedGlobalPadNum));
+  //  std::cout << " " << int(digitPos.getPadPos().getPad()) << " " << int(shiftedDigitPos.getPadPos().getPad()) << " " << int(digitPos.getGlobalPadPos().getRow()) << " " << int(shiftedDigitPos.getGlobalPadPos().getRow()) << "\n";
+
+  return {shiftedDigitPos, computePadResponse(offsetX, offsetY, gemstack)};
+}
+
+void SignalInduction::getPadResponse(const GlobalPosition3D posEle, const DigitPos digitPos, std::vector<PadResponse>& signalArray, const PadResponseMode& mode) const
+{
+  signalArray.resize(0);
+  switch (mode) {
+    case PadResponseMode::FullPadResponse: {
+      std::vector<double> offset = {{-1, 0, 1}};
```

Comment:
yes, essentially we take a look around the center of the avalanche, and since the PRF is anyways mostly projective +/-1 is more than sufficient. even for the IROC pads

Reply:
can still be `const`